core/basejob.py

```diff
@@ -28,7 +30,27 @@
 __all__ = ["SingleJob", "MultiJob"]
 
 
-class Job:
+def _fail_on_exception(func):
+    """Decorator to wrap a job method and mark the job as failed on any exception."""
+
+    def wrapper(self: "Job", *args, **kwargs):
+        try:
+            return func(self, *args, **kwargs)
+        except:
```

Comment:
Are we sure we want a bare `except` here? This will also capture things like `KeyboardInterrupt`, probably making ctrl+c somewhat unreliable. It will usually stop the script, but not if it happens to be inside of the `try` block when you press it. `SystemExit` will also be captured here ...

See: https://docs.python.org/3/library/exceptions.html#exception-hierarchy